test/contrib/test_module.py

```diff
@@ -311,3 +314,535 @@ def model():
     guide = AutoDelta(model)
     svi = SVI(model, guide, numpyro.optim.Adam(0.01), Trace_ELBO())
     svi.run(random.PRNGKey(100), 10)
+
+
+def nnx_model_by_shape(x, y):
+    from flax import nnx
+
+    class Linear(nnx.Module):
+        def __init__(self, din, dout, *, rngs):
+            self.w = nnx.Param(jax.random.uniform(rngs.params(), (din, dout)))
+            self.bias = nnx.Param(jnp.zeros((dout,)))
+
+        def __call__(self, x):
+            # Handle different Python versions by accessing the value attribute if needed
+            w_val = self.w.value if hasattr(self.w, "value") else self.w
+            bias_val = self.bias.value if hasattr(self.bias, "value") else self.bias
+            return x @ w_val + bias_val
+
+    # Pass input_shape separately - it will be handled properly by nnx_module
+    nn = nnx_module("nn", Linear, din=100, dout=100, input_shape=(100,))
+    mean = nn(x)
+    numpyro.sample("y", numpyro.distributions.Normal(mean, 0.1), obs=y)
+
+
+def nnx_model_by_kwargs(x, y):
+    from flax import nnx
+
+    class Linear(nnx.Module):
+        def __init__(self, din, dout, *, rngs):
+            self.w = nnx.Param(jax.random.uniform(rngs.params(), (din, dout)))
+            self.bias = nnx.Param(jnp.zeros((dout,)))
+
+        def __call__(self, x):
+            # Handle different Python versions by accessing the value attribute if needed
+            w_val = self.w.value if hasattr(self.w, "value") else self.w
+            bias_val = self.bias.value if hasattr(self.bias, "value") else self.bias
+            return x @ w_val + bias_val
+
+    # Directly initialize with dimensions
+    input_dim = x.shape[0]
+    # Don't pass x directly to nnx_module's constructor
+    nn = nnx_module("nn", Linear, din=input_dim, dout=100)
+    mean = nn(x)
+    numpyro.sample("y", numpyro.distributions.Normal(mean, 0.1), obs=y)
+
```

Comment:
I guess this is for the old implementation.

Reply:
yes! 🙏  removed it! https://github.com/pyro-ppl/numpyro/pull/1990/commits/1716f2d66f6af01b3ca99c9dbcf82032ef729e3b